src/main/java/hudson/plugins/gradle/injection/MavenInjectionEnvVarsRunListener.java

```diff
@@ -0,0 +1,64 @@
+package hudson.plugins.gradle.injection;
+
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.maven.AbstractMavenBuild;
+import hudson.model.FreeStyleBuild;
+import hudson.model.InvisibleAction;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.model.listeners.RunListener;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+
+import java.io.IOException;
+
+import static hudson.plugins.gradle.injection.MavenInjectionAware.JENKINSGRADLEPLUGIN_MAVEN_OPTS_PREPARED;
+import static hudson.plugins.gradle.injection.MavenInjectionAware.JENKINSGRADLEPLUGIN_MAVEN_PLUGIN_CONFIG_EXT_CLASSPATH_PREPARED;
+
+@Extension
+public class MavenInjectionEnvVarsRunListener extends RunListener<Run<?, ?>> {
```

Comment:
Maybe rename it to `MavenInjectionRunListener`? To expose fewer implementation details and to be more aligned with the `MavenInjectionEnvironmentContributor` in terms of naming.

Reply:
The purpose can be described in the javadoc.

Reply:
Done